stl/inc/algorithm

```diff
@@ -9320,6 +9333,114 @@ namespace ranges {
 #endif // _HAS_CXX20
 #endif // _HAS_CXX17
 
+template <class _RanIt, class _Pr>
+_CONSTEXPR20 void _Swap_med5_unchecked(_RanIt _First, _RanIt _Dest, _Pr _Pred) {
+    _RanIt _Guess = _First;
+    ++_Guess;
+    _RanIt _Third = _Guess;
+    ++_Third;
+    _STD _Med3_unchecked(_First, _Guess, _Third, _Pred);
+
+    _RanIt _Fourth = _Third;
+    ++_Fourth;
+    _RanIt _Fifth = _Fourth;
+    ++_Fifth;
+    if (_DEBUG_LT_PRED(_Pred, *_Fifth, *_Fourth)) {
+        swap(*_Fourth, *_Fifth); // intentional ADL
+    }
+
+    // we have _First <= _Guess (second) <= _Third and _Fourth <= _Fifth
+    if (_DEBUG_LT_PRED(_Pred, *_Fifth, *_Guess)) {
+        if (_DEBUG_LT_PRED(_Pred, *_First, *_Fifth)) {
+            _Guess = _Fifth;
+        } else {
+            _Guess = _First;
+        }
+    } else if (_DEBUG_LT_PRED(_Pred, *_Guess, *_Fourth)) {
+        if (_DEBUG_LT_PRED(_Pred, *_Third, *_Fourth)) {
+            _Guess = _Third;
+        } else {
+            _Guess = _Fourth;
+        }
+    }
+
+    swap(*_Dest, *_Guess); // intentional ADL
+}
+
+template <class _RanIt, class _Pr>
+_CONSTEXPR20 void _Nth_element_median_of_medians(_RanIt _First, _RanIt _Nth, _RanIt _Last, _Pr _Pred) {
+    using _Diff = _Iter_diff_t<_RanIt>;
+    for (;;) {
+        auto _Length = _Last - _First;
+        if (_Length <= _ISORT_MAX) {
+            _STD _Insertion_sort_unchecked(_First, _Last, _Pred); // sort any remainder
+            return;
+        }
+
+        _Diff _Median_count = _Length / static_cast<_Diff>(5);
+        auto _Median_last   = _First + _Median_count;
+        auto _Current       = _First;
+        for (auto _Median_dest = _First; _Median_dest != _Median_last; ++_Median_dest) {
+            _STD _Swap_med5_unchecked(_Current, _Median_dest, _Pred);
+            _Current += static_cast<_Diff>(5);
+        }

```

Comment:
This bit took me a bit to understand because I did not expect the medians to be collected by putting them in positions `_Current`, then `_Current + 1`, `_Current + 2`, and so on until `_Current + _Median_count`. I assumed the medians would be placed in their "expected position" if the array were sorted; that's on me :-) .

 _A subjective nit_ based on the above - I'd appreciate a short comment at the beginning of this block indicating that we'll accumulate the medians "at the beginning of the container". Perhaps @StephanTLavavej can help me improve the phrasing here, in case it can be made more accurate.